apps/docs/pages/guides/getting-started/quickstarts/kotlin.mdx

```diff
@@ -142,154 +144,58 @@ export const meta = {
 
   <StepHikeCompact.Step step={7}>
 
-    <StepHikeCompact.Details title="Create a domain object">
-      This kind of object will be consumed by the view.
-    </StepHikeCompact.Details>
-
-    <StepHikeCompact.Code>
-      ```kotlin
-      data class Product(
-          val productId: String,
-          val name: String,
-          val description: String,
-          val price: Double,
-          val image: String,
-          val category: String,
-          val nutrition: String,
-          val _id: Int,
-      )
-      ```
-
-    </StepHikeCompact.Code>
-
-  </StepHikeCompact.Step>
-
-  <StepHikeCompact.Step step={8}>
-
     <StepHikeCompact.Details title="Query data from the app">
-      Create a repository to interact with the data source.
+      Use `LaunchedEffect` to fetch data from the database and display it in a `LazyColumn`.
+
+      Replace the default `MainActivity` class with the following code.
     </StepHikeCompact.Details>
 
     <StepHikeCompact.Code>
       ```kotlin
-      interface ProductRepository {
-        fun getProducts(): List<ProductDto>
-      }
-
-      class ProductRepositoryImpl @Inject constructor(
-            private val postgrest: Postgrest,
-        ) : ProductRepository {
-          override suspend fun getProducts(): List<ProductDto> {
-            val result = client.postgrest["products"]
-                .select().decodeList<ProductDto>()
-            // Handle result data for next step
-            return result
+      class MainActivity : ComponentActivity() {
+          override fun onCreate(savedInstanceState: Bundle?) {
+              super.onCreate(savedInstanceState)
+              setContent {
+                  SupabaseTutorialTheme {
+                      // A surface container using the 'background' color from the theme
+                      Surface(
+                          modifier = Modifier.fillMaxSize(),
+                          color = MaterialTheme.colorScheme.background
+                      ) {
+                          CountriesList()
+                      }
+                  }
+              }
```

Comment:
@jan-tennert 
Could you check if I'm not doing anything totally weird here? 

I know this quick start was contributed by someone else, but I always wanted to make it simpler so that people can interact with Supabase on Android as quickly as possible without worrying too much about state management and what not.

Reply:
Thanks for the comment. I have added a little note on the details section of the step.